### PR TITLE
Fix and re-enable newer mitmproxy usage in tests.

### LINKED ISSUE #5454: test_proxy_connect.py hangs with mitmproxy 8.0.0 (affecting 3.8 tests)
mitmproxy 8.0.0 was released 2 days ago, it's likely to be the cause of this.

Also it looks like we currently don't install mitmproxy at all for 3.9+.

### PR DESCRIPTION
**Pull request by wRAR**
As usual, a two-character fix.

Fixes #5454.

### CODE REVIEW

File: tests/test_proxy_connect.py [hunk 1]
```diff
@@ -46,7 +47,7 @@ def start(self):
             stdout=PIPE,
         )
         line = self.proc.stdout.readline().decode("utf-8")
-        host_port = re.search(r"listening at http://([^:]+:\d+)", line).group(1)
+        host_port = re.search(r"listening at (?:http://)?([^:]+:\d+)", line).group(1)
```
Comment (wRAR) [inline tests/test_proxy_connect.py:50]: The output format has changed in some newer version but we can support both.

File: tests/test_proxy_connect.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def start(self):
         self.proc = Popen(
             [
                 sys.executable,
+                "-u",
```
Comment (wRAR) [inline tests/test_proxy_connect.py:34]: mitmproxy changed the code that prints log lines and we now need to disable stdout buffering to get it via Popen.

File: tox.ini [hunk 3]
```diff
@@ -11,11 +11,7 @@ minversion = 1.7.0
 deps =
     -rtests/requirements.txt
     # mitmproxy does not support PyPy
-    # Python 3.9+ requires mitmproxy >= 5.3.0
-    # mitmproxy >= 5.3.0 requires h2 >= 4.0, Twisted 21.2 requires h2 < 4.0
-    #mitmproxy >= 5.3.0; python_version >= '3.9' and implementation_name != 'pypy'
-    # The tests hang with mitmproxy 8.0.0: https://github.com/scrapy/scrapy/issues/5454
-    mitmproxy >= 4.0.4, < 8; python_version < '3.9' and implementation_name != 'pypy'
+    mitmproxy; implementation_name != 'pypy'
```
Comment (kmike) [inline tox.ini:14]: is there a minimal mitmproxy version, like `>8`?

**Review by kmike — APPROVED** [review]

File: tox.ini (hunk 3 above)
Comment (wRAR) [reply tox.ini:14]: The code should work with any mitmproxy version.

**Pull request merged**